java/arcs/core/util/UnionFind.kt

```diff
@@ -13,18 +13,30 @@ package arcs.core.util
 
 /**
  * A union-find datastructure for computing equivalence classes consisting of [E] instances.
- * This datastructure is not thread safe.
+ * Optionally, a value of type [I] may be associated with the equivalence class. If the client
+ * does not need to associated additional information with the class, they may use the [Unit]
+ * type for [I]. This datastructure is not thread safe.
  */
-class UnionFind<E> {
+class UnionFind<E, I> {
```

Comment:
100% Optional: You might be able to make a typealias to make it easier to build an instance without additional information:

```kotlin
typealias UnionFind<E> = UnionFind<E, Unit>
```

Reply:
Also, you could make extension functions for the typealias that wrap the functions defined on the main UnionFind, passing `Unit` for the `I`-typed param.

Reply:
I tried this, but got a redeclaration error. Typealias cannot have the same name as the class? See https://discuss.kotlinlang.org/t/allow-typealias-to-have-the-same-name-as-generic-class/6789/5 

What we want is something like as mentioned in the discussion: 
```
class Foo<T : Bar = Baz>
```